### DISCUSSION TITLE
Is there a selected-row like syntax for modi?

### DISCUSSION THREAD

**Discussion by s0la** [davatorium/rofi#1888] [Q&A] [answered]
Hey guys, I guess title says it all..
I know there are certain options that can be applied using "\0<option>\x1f<value>" syntax, e.g.

\0active\x1f5

which will make 5th item marked as active..

I'm wondering if there is a way to set selected item for modi (same way -selected-row does it for dmenu), using above syntax..

Thanks! :)

**Comment by DaveDavenport** [comment]
What modi are we talking about? There are many.
        • window
        • windowcd
        • run
        • ssh
        • drun
        • combi
        • keys
        • filebrowser
        • recursivebrowser
        • script

**Reply by s0la** [reply]
I'm using it with a custom script, so I guess that means script.. I'm using it like this:

rofi -modi "<name>:/path/to/my/script.sh" -show <name>

So in my script, where I generate "menu content", I'd like to be able to specify which menu item should be selected by default, like I'm doing with the active item (example I gave in initial message)..

Thanks!

**Reply by DaveDavenport** [reply]
I think keep-selection and new-selection (https://davatorium.github.io/rofi/1.7.5/rofi-script.5/#passing-mode-options) might do what you want.

**Reply by s0la** [reply]
Great, sounds exactly like what I was looking for.. Is this the right way to use it?

printf "\0keep-selection\x1ftrue\0new-selection\x1f2\n"

where 2 near the end is an item index.. I tried that but it still defaults to the first item..

**Reply by DaveDavenport** [reply]
This was my test program (see source repo):
```bash
#!/usr/bin/env bash

if [ "$*" = "quit" ]
then
    exit 0
fi

if [ "$@" ]
then
    # Override the previously set prompt.
    echo -en "\0theme\x1felement-text { background-color: $@;}\n"
    echo -en "\0keep-selection\x1ftrue\n"
    echo -en "\0new-selection\x1f2\n"
    echo "red"
    echo "lightgreen"
    echo "lightblue"
    echo "lightyellow"
    echo "quit"
else
    echo -en "\x00prompt\x1ftesting\n"
    echo -en "\0urgent\x1f0,2\n"
    echo -en "\0active\x1f1\n"
    echo -en "\0keep-selection\x1ftrue\n"
    echo -en "\0message\x1fSpecial <b>bold</b>message\n"

    echo "red"
    echo "lightgreen"
    echo "lightblue"
    echo "lightyellow"
    echo "pink"
    echo "green"
    echo "blue"
    echo "gold"
    echo "quit"
fi
```

**Reply by s0la** [reply]
This works as described after selecting an item, but doesn't work on initial menu (when app is started for the first time)..
What I was hoping to achieve is basically this:

echo -e '1\n2\n3\n4\n5' | rofi -dmenu -selected-row 2

but using modi instead, with rofi's specific syntax within the script..
So basically starting the menu with preselected item.. Script/app will close immediately after selecting an item, so "keeping" selection for the next "iteration" is not really an option..
If this is not possible it's not a big deal, I'm just curious since it would come handy for this usecase.. That's why I opened this as a question rather than feature request :)

**Reply by DaveDavenport** [reply]
Yeah, that does not work, as there is no guarantee you actually start in that mode. 

```bash
rofi -show drun -modes 'drun,my:./myscript.sh' 
```

**Reply by DaveDavenport** [reply]
(on startup you can use -selected-row also with other modes)

**Reply by s0la** [reply]
I don't think that would work since row is supposed to be "calculated" dynamically within the script..
To provide a bit more context, it's a custom script for workspace switching, supporting a couple of different actions..

[
![ws_rofi](https://github.com/davatorium/rofi/assets/6886981/08754e64-77a4-4cc1-829b-f5c0e93ff56a)
](url)

So the point was to get current workspace within the script and set it as a default row, since I think it's easier to navigate/traverse the list starting from the relative/current row rather than the first one.. Granted, in this example 4 row is a really small number, so it's basically insignificant, but since the workspace number is not limited and can grow, it would be generally more convenient approach..

I could use drun method with selected-row and provide workspace value using the subshell, but that leaves me with drun menu which has nothing to do with workspaces :)

But as I said, it doesn't really matter, I already took enough of your time, it's just a small convenience, not the deal-breaking feature..
Anyways, thanks a lot for your willingness to help me make this work, I really appreciate it!!

Btw, since I already attached the image, I have another question (way simpler though :) )
As you can see, I'm using font icons for those vertically aligned buttons on the right with the red background, but I'd prefer if those icons could be a bit larger than the list item font.. So my question is is there a way to set different font size for different "components" (e.g. modeswitcher and list)?

Also, if you think it's more appropriate, I could open this as a separate question, since it's not related to the original subject..
I'm just asking here because I think it's kinda simple yes or no question, but can open a new one if that's what you prefer :)

**Reply by DaveDavenport** [reply, marked as answer]
> So my question is is there a way to set different font size for different "components" (e.g. modeswitcher and list)?

yes, you can set a font on any widget.

**Reply by s0la** [reply]
Damn, I could swear I tried that but wasn't able to make it work, otherwise I wouldn't ask something I can easily try and answer myself..
Anyway, it works perfectly, thanks a ton! :)
